core/src/main/java/org/ehcache/core/statistics/DefaultStatisticsService.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core.statistics;
+
+import org.ehcache.Cache;
+import org.ehcache.Status;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.core.InternalCache;
+import org.ehcache.core.events.CacheManagerListener;
+import org.ehcache.core.spi.service.CacheManagerProviderService;
+import org.ehcache.core.spi.service.StatisticsService;
+import org.ehcache.core.spi.store.InternalCacheManager;
+import org.ehcache.core.spi.store.Store;
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceDependencies;
+import org.ehcache.spi.service.ServiceProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terracotta.management.model.capabilities.descriptors.StatisticDescriptor;
+import org.terracotta.management.model.stats.Statistic;
+import org.terracotta.management.model.stats.StatisticRegistry;
+import org.terracotta.management.model.stats.StatisticType;
+import org.terracotta.statistics.MappedOperationStatistic;
+import org.terracotta.statistics.OperationStatistic;
+import org.terracotta.statistics.StatisticsManager;
+import org.terracotta.statistics.derived.OperationResultFilter;
+import org.terracotta.statistics.derived.latency.DefaultLatencyHistogramStatistic;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+import static org.ehcache.core.statistics.StatsUtils.findOperationStatisticOnChildren;
+import static org.terracotta.statistics.StatisticBuilder.operation;
+
+/**
+ * Default implementation using the statistics calculated by the observers set on the caches.
+ */
+@ServiceDependencies(CacheManagerProviderService.class)
+public class DefaultStatisticsService implements StatisticsService, CacheManagerListener {
```

Comment:
We need to eliminate all the static methods from this class and add them to the StatisticsService interface. This way we don't enforce the statistics library static linking method on the StatisticsService.